### PR TITLE
[Micromamba] Add `env update`

### PR DESCRIPTION
**Pull request by Hind-M**
Fix part of #2819

### CODE REVIEW

File: libmamba/src/api/update.cpp [hunk 1]
```diff
@@ -127,6 +127,24 @@ namespace mamba
...
+                std::vector<std::string> remove_specs;
+                for (auto& it : hist_map)
+                {
+                    if (std::find(update_specs.begin(), update_specs.end(), it.second.name)
+                        == update_specs.end())
+                    {
+                        remove_specs.push_back(it.second.name);
+                    }
+                }
+                solver.add_jobs(remove_specs, SOLVER_ERASE | SOLVER_CLEANDEPS);  // TODO it doesn't
```
Comment (Hind-M) [inline libmamba/src/api/update.cpp:142]: The wanted behavior here is to remove `remove_specs` **and** their dependencies, but only the specs are removed here.
Adding `SOLVER_CLEANDEPS` doesn't seem to be working... 
Any ideas @AntoinePrv ?

Comment (AntoinePrv) [reply libmamba/src/api/update.cpp:142]: Does it work with only `SOLVER_ERASE`?

Comment (Hind-M) [reply libmamba/src/api/update.cpp:142]: Yes, adding `SOLVER_CLEANDEPS` doesn't change anything on that matter (I just left it for now to illustrate).

Comment (Hind-M) [reply libmamba/src/api/update.cpp]: I think that it was just some mixing with the installation steps locally when I first tested it but is working fine now with the `micromamba` tests 👍

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**